platform-sdk/docs/core/dynamic-address-book/DynamicAddressBook-PlatformDesign.md

```diff
@@ -0,0 +1,107 @@
+# Dynamic Address Book - Platform Design
+
+![Dynamic Address Book Platform Design](DynamicAddressBook-PlatformDesign.drawio.svg)
+
+## Network Connection Effects
+
+Assumption: That the network is fully connected, all nodes connect to each other.
+
+* Need to validate that this is true.
+* Huge gap in testing if not.
+
+### Add Node
+
+Existing nodes need to create or receive mutual TLS gossip connections with the added node.
+
+This change takes effect when a node is added in the `currentEffectiveRoster` for the `pendingConsensusRound`.
+
+### Remove Node
+
+Existing nodes need to stop gossiping with the removed node. Events from the removed node can still be gossiped,
+validated and come to consensus until the round this new roster takes effect becomes ancient.
+
+This change takes effect when a node is removed from the `currentEffectiveRoster` for the `pendingConsensusRound`.
+
+### Rotate Signing Key
+
+All nodes need to tear down their existing mutual TLS connection with the node that has had its key rotated and
+establish a new connection with the node using the new signing certificate as the trusted root certificate.
+
+This change takes effect when a node's key changes in the `currentEffectiveRoster` for the `pendingConsensusRound`.
+
+### Node Weight Change
+
+This has no effect on the existence of network connections.
+
+## Event Creation Effects
+
+### Add Node
+
+Begin using events from the node as parents of new events.
+
+This change takes effect when a node is added to the `currentEffectiveRoster` for the `pendingConsensusRound`.
+
+This change will modify consensus weight for the network and the event creation algorithm will be reset.
```

Comment:
Why does this modify weight for the network? A node can join as zero-weight.

Reply:
What does reseting the algorithm mean?

Reply:
One could view the addition of a node as an ontological change to the distribution of the weight.  True the total weight of the network is unchanged and the value of weight distributed to nodes is numerically the same as before but because there exists a new node, its a different distribution.  

Mathematically a Function is defined on a Domain.  We have a function mapping weight to nodes.   If the Domain of the function changes, one could argue the new function is a different function.  

@cody-littley, thoughts?

Reply:
If we are buffering newer rounds than `pendingConsensusRound` and caching the latest tips in each newer round, then when the round is updated and event creation advances to the next `pendingConsensusRound` with its `currentlyEffectiveRoster` the set of tips the event creator is using needs to be updated with the cached tips in the round coming off the buffer.  All the nodes are given the weight distribution in the new `currentlyEffectiveRoster`.   I think this should happen regardless of whether or not the weight actually changed per node.   Safer to reset than to patch the data.

Reply:
There are two steps to resetting the tipset event creator. It's not quite as simple as just clearing everything, but it's also not super complex.

1) for all events being tracked (i.e non-ancient events), update the tipset to include the proper nodes. Each tipset is stored as an array.
2) reset the snapshot to be equal to the tipset of the most recently created event.

Step 1 can be skipped for consensus weight only changes.

We need to take these actions regardless of whether or not a new node has zero stake. This is because the tipset algorithm is willing to use zero stake events as parents.